### ISSUE TITLE
[question] How can I access a JSON key without knowing it beforehand?

### ISSUE THREAD

**Issue by miguelvr** [ovh/venom#260]
I have a JSON response that returns some hashes in a JSON object. I would like to access one of these keys to use for comparison in another step of my test suite.

Example: 

```json
{
  "claims": {
    "aaaaaa": {...},
    "bbbbbb": {...}
  }
}
```

**Comment by gwleclerc** [comment]
Hi, 
I have an example that extracts a variable from a result and uses it in another test.
https://github.com/Thiht/smocker/blob/master/tests/features/locks_mocks.yml#L26

I hope it will help you.

**Comment by miguelvr** [comment]
I don't think that covers my use case. I don't know if I was explicit enough, let me try again.

I have a JSON response and I need to retrieve the key based on a property of its contents, in this particular case it's JWT claims.

e.g.

```json
{
  "claims": {
    "hash1": {
      "endswith": "123"
    },
    "hash2": {
      "endswith": "789"
    }
  }
}
```

In this case I want to know which key (hash1 or hash2) contains "endswith" == "789", which would be "hash2"

**Comment by yesnault** [comment]
So, you want to know which attribute hash1 or hash2 endswith 789, to probably re-use the attribute on further step. You can do that, not "natively" with venom, but something like that:
1 step http, to Call HTTP, and put the raw body in a file
1 step script to grep in the file the good attribute, then export the variable as a new venom variable
1 other step to re-use this variable

Example of extract variable on the fly from results and reuse it in step after

```yml
name: MyTestSuite
testcases:
- name: testA
  steps:
  - type: exec
    script: echo 'foo with a bar here'
    vars:
      myvariable:
        from: result.systemout
        regex: foo with a ([a-z]+) here

- name: testB
  steps:
  - type: exec
    script: echo {{.testA.myvariable}}
    assertions:
    - result.code ShouldEqual 0
    - result.systemout ShouldContainSubstring bar
```
